src/main/java/org/broadinstitute/hellbender/engine/FeatureContext.java

```diff
@@ -107,11 +110,77 @@ public SimpleInterval getInterval() {
      *         this FeatureContext's query interval. Empty List if there is no backing data source and/or interval.
      */
     public <T extends Feature> List<T> getValues( final FeatureInput<T> featureDescriptor ) {
-        if ( featureManager == null || interval == null ) {
+        return getValues(featureDescriptor, interval);
+    }
+
+    /**
+     * Gets all Features from the source represented by the provided FeatureInput argument that overlap
+     * this FeatureContext's query interval. Will return an empty List if this FeatureContext has
+     * no backing source of Features and/or interval.
+     *
+     * Returned Features are not guaranteed to be in any particular order.
+     *
+     * @param featureDescriptor FeatureInput argument for which to fetch Features
+     * @param <T> type of Feature in the data source backing the provided FeatureInput
+     * @param windowLeadingBases Number of extra reference bases to include before the start of our interval. Must be >= 0.
+     * @param windowTrailingBases Number of extra reference bases to include after the end of our interval. Must be >= 0.
+     * @return All Features in the data source backing the provided FeatureInput that overlap
+     *         this FeatureContext's query interval. Empty List if there is no backing data source and/or interval.
+     */
+    public <T extends Feature> List<T> getValues( final FeatureInput<T> featureDescriptor, final int windowLeadingBases, final int windowTrailingBases) {
+        return getValues(featureDescriptor, getWindow(windowLeadingBases, windowTrailingBases));
+    }
+
+    private <T extends Feature> List<T> getValues( final FeatureInput<T> featureDescriptor, final SimpleInterval window) {
```

Comment:
Rename the `window` parameter here to `queryInterval`, since `window` implies extra context around the interval represented by the `interval` instance variable, but here it is just the absolute interval to query.